flytekit/common/tasks/notebook.py

```diff
@@ -0,0 +1,542 @@
+from __future__ import absolute_import
+
+import flytekit.common.interface as _interface
+from flytekit.models.interface import Variable
+import os as _os
+from flytekit.common.tasks import output as _task_output
+import json
+import papermill as pm
+from flytekit import __version__
+import importlib as _importlib
+
+import datetime as _datetime
+from pyspark import SparkConf, SparkContext
+import sys as _sys
+import six as _six
+from flytekit.bin import spark_executor
+from flytekit.common import constants as _constants
+from flytekit.common.exceptions import scopes as _exception_scopes
+from flytekit.common.tasks import sdk_runnable as _sdk_runnable
+from flytekit.common.tasks import spark_task as _spark_task
+
+from flytekit.common.types import helpers as _type_helpers
+from flytekit.models import literals as _literal_models, task as _task_models
+from google.protobuf.json_format import MessageToDict as _MessageToDict
+from flytekit.sdk.types import Types
+
+from flytekit.common.types.helpers import pack_python_std_map_to_literal_map as _packer
+from flytekit.common.types import primitives as _p
+from google.protobuf import text_format
+from flytekit.common import sdk_bases
+from flytekit.common.tasks import task as base_tasks
+from flytekit.engines import loader as _engine_loader
```

Comment:
nit: for all imports, let's import the module itself and not the leaf attribute. it makes it easier to refactor things.  

2. If possible, use commas if they have the same package.  example:

```
from flytekit.common.tasks import sdk_runnable as _sdk_runnable, spark_task as _spark_task
```

3.  Always prefix anything meant to be private to this package with underscore to prevent it leaking into other namespaces.

Reply:
done, I might have missed some but hopefully got them all.